frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/withMergeNodes.ts

```diff
@@ -1,42 +1,47 @@
 import { ReactEditor } from 'slate-react';
-import { Editor, Element, NodeEntry, Node, Transforms, Point } from 'slate';
-import { CustomEditor } from '$app/components/editor/command';
+import { Editor, Element, NodeEntry, Node, Transforms } from 'slate';
 
 export function withMergeNodes(editor: ReactEditor) {
   const { mergeNodes } = editor;
 
-  // before merging nodes, check whether the node is a block and whether the selection is at the start of the block
-  // if so, move the children of the node to the previous node
   editor.mergeNodes = (...args) => {
-    const { selection } = editor;
-
+    mergeNodes(...args);
     const isBlock = (n: Node) =>
       !Editor.isEditor(n) && Element.isElement(n) && n.type !== undefined && n.level !== undefined;
 
-    const [match] = Editor.nodes(editor, {
+    const [merged] = Editor.nodes(editor, {
       match: isBlock,
     });
 
-    if (match && selection) {
-      const [node, path] = match as NodeEntry<Element>;
-      const start = Editor.start(editor, path);
-
-      if (Point.equals(selection.anchor, start)) {
-        const previous = Editor.previous(editor, { at: path });
-        const [previousNode] = previous as NodeEntry<Element>;
-        const previousLevel = previousNode.level ?? 1;
+    if (!merged) {
+      return;
+    }
 
-        const children = CustomEditor.findNodeChildren(editor, node);
+    const [mergedNode, path] = merged as NodeEntry<Element & { level: number }>;
 
-        children.forEach((child) => {
-          const childPath = ReactEditor.findPath(editor, child);
+    const nextNode = editor.children[path[0] + 1] as Element & { level: number };
 
-          Transforms.setNodes(editor, { level: previousLevel + 1, parentId: previousNode.blockId }, { at: childPath });
-        });
-      }
+    if (mergedNode.blockId === nextNode.parentId) {
+      return;
     }
 
-    mergeNodes(...args);
+    // check if the old node is removed
+    const oldNodeRemoved = !editor.children.some((child) => (child as Element).blockId === nextNode.parentId);
```

Comment:
```ts
if (oldNodeRemoved) {
  const newLevel = mergedNode.level + 1;
  editor.children.forEach((child, index) => {
    if ((child as Element).parentId === nextNode.parentId) {
      Transforms.setNodes(
        editor,
        { level: newLevel, parentId: mergedNode.blockId },
        { at: [index] } 
      );
    }
  });
}
```
Can I combine two loop into one? It can improve efficiency